Cabal/Distribution/Simple/Register.hs

```diff
@@ -230,13 +244,32 @@ invokeHcPkg verbosity comp conf dbStack extraArgs =
 withHcPkg :: String -> Compiler -> ProgramConfiguration
           -> (HcPkgInfo -> IO a) -> IO a
 withHcPkg name comp conf f =
-  case compilerFlavor comp of
-    GHC   -> f (GHC.hcPkgInfo conf)
-    GHCJS -> f (GHCJS.hcPkgInfo conf)
-    LHC   -> f (LHC.hcPkgInfo conf)
+  case getHcPkgInfo comp conf of
+    Just hcPkgInfo -> f hcPkgInfo
     _     -> die ("Distribution.Simple.Register." ++ name ++ ":\
                   \not implemented for this compiler")
 
+getHcPkgInfo comp conf = case compilerFlavor comp of
```

Comment:
This needs a type signature, otherwise the build fails on Travis due to `-Wall`:

```
    Top-level binding with no type signature:

      getHcPkgInfo :: Compiler -> ProgramConfiguration -> Maybe HcPkgInfo
```